modules/core/src/v2/pendingApproval.ts

```diff
@@ -307,6 +315,29 @@ export class PendingApproval {
     return await this.reject(params);
   }
 
+  /**
+   * Recreate and sign TSS transaction
+   * @param {ApproveOptions} params needed to get txs and use the walletPassphrase to tss sign
+   * @param {RequestTracer} reqId id tracer.
+   */
+  async recreateAndSignTSSTransaction(params: ApproveOptions, reqId: RequestTracer): Promise<any> {
+    const { txRequestId, walletPassphrase } = params;
+
+    if (!this.wallet) {
+      throw new Error('Wallet not found');
+    }
+
+    if (!walletPassphrase) {
+      throw new Error('walletPassphrase not found');
+    }
+
+    if (!txRequestId) {
+      throw new Error('txRequestId not found');
+    }
+
+    return this.tssUtils.recreateTxRequest(txRequestId, walletPassphrase, reqId);
```

Comment:
you want to pass in the user's decrypted prv instead of wallet passphrase:

Reply:
discussed on chat, thank you for the help.